### PR TITLE
Do not overwrite error 'metadata' if already specified by caller.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/grpc/grpc-node/issues/1236

Not certain the best way to add tests for this change.  Would appreciate assistance as I have only submitted one other (non-runtime) PR for this repo.  Thanks!

### CODE REVIEW

File: packages/grpc-js/src/server-call.ts [hunk 1]
```diff
@@ -455,7 +455,9 @@ export class Http2ServerCallStream<
     }
 
     if (err) {
-      err.metadata = metadata;
+      if (!err.hasOwnProperty('metadata')) {
+        err.metadata = metadata;
+      }
```
Comment (CyrusNajmabadi) [inline packages/grpc-js/src/server-call.ts]: Open questions:
1. should i comment this?
2. would it be more appropriate to merge the existing `.metadata` property and the `metadata` local?  Note that the native impl does *not* do this and only uses the `err.metadata` property alone: https://github.com/grpc/grpc-node/blob/fa20c5e565439a6131508fcce782fdd15946f394/packages/grpc-native-core/src/server.js#L62-L65

**Comment by murgatroid99** [comment]
The best place to put tests for this is in `test/api`, so that the same test is run against both implementations. It would be reasonable add this test to `test/api/error_test.js`.

**Comment by CyrusNajmabadi** [comment]
> The best place to put tests for this is in test/api, so that the same test is run against both implementations. It would be reasonable add this test to test/api/error_test.js.

Great.  I'll try to figure out how to do that!

**Comment by CyrusNajmabadi** [comment]
How do i go about running tests btw?

File: packages/grpc-js/src/server-call.ts (hunk 1 above)
Comment (murgatroid99) [reply packages/grpc-js/src/server-call.ts]: 1. A comment probably isn't necessary. I think it's clear what's happening here.

2.  The behavior here should just match what the native implementation does.

**Comment by murgatroid99** [comment]
If you `npm install` in the root directory, `packages/grpc-native-core`, and `packages/grpc-js`, you can run the tests by running `gulp test` (or `../node_modules/.bin/gulp test`) in the `test` directory. Alternatively, you can just push the change and we can run it on our CI.

**Comment by CyrusNajmabadi** [comment]
> Alternatively, you can just push the change and we can run it on our CI.

Awesome.  Working on test now.

**Comment by CyrusNajmabadi** [comment]
Unfortunately, getting: 

```
[15:12:13] Requiring external module ts-node/register
[15:12:14] Using gulpfile ~/go/src/github.com/CyrusNajmabadi/grpc-node/test/gulpfile.ts
[15:12:14] Starting 'test'...
Running /home/cyrusn/go/src/github.com/CyrusNajmabadi/grpc-node/test/api/*.js with native server + native client
fs.js:114
    throw err;
    ^

Error: ENOENT: no such file or directory, open '/home/cyrusn/go/src/github.com/CyrusNajmabadi/grpc-node/packages/grpc-native-core/deps/grpc/etc/roots.pem'
```

File: test/api/error_test.js [hunk 2]
```diff
@@ -562,5 +562,131 @@ describe(`${anyGrpc.clientName} client -> ${anyGrpc.serverName} server`, functio
...
+      after(function() {
+        server.forceShutdown();
+      });
+      describe('Existing metadata', function() {
+        it('should not be present when a unary call succeeds', function(done) {
+          var call = client.unary({error: false}, function(err, data) {
+            assert.ifError(err);
+          });
+          call.on('status', function(status) {
+            assert.true(status.metadata.get('existing-metadata', []));
```
Comment (murgatroid99) [inline test/api/error_test.js]:
```suggestion
            assert.deepStrictEqual(status.metadata.get('existing-metadata'), []);
```

File: test/api/error_test.js [hunk 3]
```diff
@@ -562,5 +562,131 @@ describe(`${anyGrpc.clientName} client -> ${anyGrpc.serverName} server`, functio
...
+            done();
+          });
+        });
+        it('should be present when a unary call fails', function(done) {
+          var call = client.unary({error: true}, function(err, data) {
+            assert(err);
+          });
+          call.on('status', function(status) {
+            assert.deepEqual(status.metadata.get('existing-metadata'), ['yes']);
```
Comment (murgatroid99) [inline test/api/error_test.js]:
```suggestion
            assert.deepStrictEqual(status.metadata.get('existing-metadata'), ['yes']);
```

**Comment by murgatroid99** [comment]
Thank you for your contribution.

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**